system-tests/lib/cre/don/config/config.go

```diff
@@ -50,7 +51,27 @@ func Generate(input types.GenerateConfigsInput, factoryFns []types.ConfigFactory
 
 	// prepare chains, we need chainIDs, URLs and selectors to get contracts from AddressBook
 	workerEVMInputs := make([]*WorkerEVMInput, 0)
+	workerSolanaInputs := make([]*WorkerSolanaInput, 0)
 	for chainSelector, bcOut := range input.BlockchainOutput {
+		if bcOut.SolChain != nil {
+			// Since solana chainID is base58 encoded genesis hash, we can't pass chainID directly from yaml.
+			client, ok := input.SolClients[chainSelector]
+			if !ok {
+				return nil, errors.Errorf("failed to find sol client for selector %d", chainSelector)
+			}
+			chainID, err := client.GetGenesisHash(context.Background())
+			if err != nil {
+				return nil, errors.Wrap(err, "failed to get chainID from solana")
+			}
+
+			workerSolanaInputs = append(workerSolanaInputs, &WorkerSolanaInput{
+				Name:    fmt.Sprintf("node-%d", bcOut.SolChain.ChainSelector),
+				ChainID: chainID.String(),
+				NodeURL: bcOut.BlockchainOutput.Nodes[0].InternalHTTPUrl,
+				// TODO add the rest solana inputs (forwarder, forwarder state from datastore)
+			})
+			continue
+		}
```

Comment:
do you think could encapsulate that as a new factory function? similarly to how it's done for the gateway. That way we could separate it more clearly, and when you initialise the factory, in the higher order function you could call the function above to get the genesis hash